### PR TITLE
All fixes

### PR DESCRIPTION
**Pull request by Ilya-Chudinovskikh**
This is a new branch without any authentication-related features.

### CODE REVIEW

File: WordOfTheDay.Repository/WordsRepository.cs [hunk 1]
```diff
@@ -22,13 +22,19 @@ private static DateTime DateToday
         }
         public async Task<WordCount> WordOfTheDay()
         {
+
             var wordOfTheDay = await _context.Words
                 .LaterThan(DateToday)
-                .FindWordOfTheDay();
-            
+                .GroupBy(word => word.Text, (text, words) => new WordCount { Word = text, Count = words.Count(word => word.Text == text) })
```
Comment (php5art) [inline WordOfTheDay.Repository/WordsRepository.cs]: Can you move this logic to extension method? And use it also for calculating close words?

File: WordOfTheDay.Repository/WordsRepository.cs [hunk 2]
```diff
@@ -22,13 +22,19 @@ private static DateTime DateToday
...
+                .OrderByDescending(w => w.Count)
+                .FirstOrDefaultAsync();
+
+            if (wordOfTheDay == null)
+                return null;
+
             return wordOfTheDay;
         }
-        public async Task<List<WordCount>> CloseWords(string email)
+        public async Task<IEnumerable<WordCount>> CloseWords(string email)
```
Comment (php5art) [inline WordOfTheDay.Repository/WordsRepository.cs]: return Task<List\<WordCount\>>

File: WordOfTheDay.Repository/IWordsRepository.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@ public interface IWordsRepository
     {
         Task<WordCount> WordOfTheDay();
 
-        Task<List<WordCount>> CloseWords(string word);
+        Task<IEnumerable<WordCount>> CloseWords(string word);
```
Comment (php5art) [inline WordOfTheDay.Repository/IWordsRepository.cs]: return Task<List\<WordCount\>>

File: WordOfTheDay.Domain/WordsServices.cs [hunk 4]
```diff
@@ -23,15 +27,15 @@ public Task<WordCount> WordOfTheDay()
             return wordOfTheDay;
         }
         
-        public Task<List<WordCount>> CloseWords(string email)
+        public Task<IEnumerable<WordCount>> CloseWords(string email)
```
Comment (php5art) [inline WordOfTheDay.Domain/WordsServices.cs]: return Task<List\<WordCount\>>

File: WordOfTheDay.Domain/WordOfTheDay.Domain.csproj [hunk 5]
```diff
@@ -17,5 +17,9 @@
 	  <_Parameter1>DynamicProxyGenAssembly2</_Parameter1>
 	</AssemblyAttribute>
   </ItemGroup>
+	
+  <ItemGroup>
+    <PackageReference Include="FreeGeoIPCore" Version="1.0.3" />
```
Comment (php5art) [inline WordOfTheDay.Domain/WordOfTheDay.Domain.csproj]: You don't need this package

File: WordOfTheDay.Domain/IWordsServices.cs [hunk 6]
```diff
@@ -10,7 +10,7 @@ public interface IWordsServices
     {
         Task<WordCount> WordOfTheDay();
 
-        Task<List<WordCount>> CloseWords(string word);
+        Task<IEnumerable<WordCount>> CloseWords(string word);
```
Comment (php5art) [inline WordOfTheDay.Domain/IWordsServices.cs]: return Task<List\<WordCount\>>

**Review by php5art — CHANGES_REQUESTED** [review]

File: WordOfTheDay.Repository/WordsRepository.cs (hunk 2 above)
Comment (Ilya-Chudinovskikh) [reply WordOfTheDay.Repository/WordsRepository.cs]: done

File: WordOfTheDay.Repository/IWordsRepository.cs (hunk 3 above)
Comment (Ilya-Chudinovskikh) [reply WordOfTheDay.Repository/IWordsRepository.cs]: done

File: WordOfTheDay.Domain/WordsServices.cs (hunk 4 above)
Comment (Ilya-Chudinovskikh) [reply WordOfTheDay.Domain/WordsServices.cs]: done

File: WordOfTheDay.Domain/WordOfTheDay.Domain.csproj (hunk 5 above)
Comment (Ilya-Chudinovskikh) [reply WordOfTheDay.Domain/WordOfTheDay.Domain.csproj]: done

File: WordOfTheDay.Domain/IWordsServices.cs (hunk 6 above)
Comment (Ilya-Chudinovskikh) [reply WordOfTheDay.Domain/IWordsServices.cs]: done

File: WordOfTheDay.Repository/WordsRepository.cs (hunk 1 above)
Comment (Ilya-Chudinovskikh) [reply WordOfTheDay.Repository/WordsRepository.cs]: There is some issue described in the Skype

Comment (Ilya-Chudinovskikh) [reply WordOfTheDay.Repository/WordsRepository.cs]: done

**Comment by Ilya-Chudinovskikh** [comment]
The last 2 commits (82c3a6d, bec298d) are made by mistake. They show the same state of the project as the previous does.

**Review by php5art — APPROVED** [review]

**Pull request merged**